src/modules/common/components/ScrollingSearchContainer.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import style from './ScrollingSearchContainer.css'
+import { animateScroll } from 'react-scroll/modules/index'
+import SearchInput from './SearchInput'
+
+export class ScrollingSearchContainer extends React.Component {
+  static propTypes = {
+    filterText: PropTypes.string.isRequired,
+    onFilterTextChange: PropTypes.func.isRequired,
+    stickyTop: PropTypes.number.isRequired,
+    spaceSearch: PropTypes.bool,
+    children: PropTypes.node
+  }
+
+  static defaultProps = {
+    stickyTop: 0
+  }
+
+  onClick = () => {
+    const scrollTop = document.documentElement.scrollTop
+    if (scrollTop < this.state.top) {
+      this.scroll()
+    }
+  }
+
+  scroll = () => {
+    animateScroll.scrollTo(this.state.top, {duration: 500})
+  }
+
+  onChange = value => {
+    this.props.onFilterTextChange(value)
+    this.scroll()
+  }
+
+  setDimensions = node => {
+    if (node) {
+      const {top, height} = node.getBoundingClientRect()
+      this.setState(Object.assign({}, this.state, {top, height}))
+    }
+  }
+
+  render () {
+    return <React.Fragment>
+      <div ref={this.setDimensions} className={style.searchBar}
+           style={{'top': this.props.stickyTop + 'px'}}>
```

Comment:
another usecase for styled components, I don't like mixing inline styling and styling in separate css sheets, but right now there is no better solution, right?